tests/foreman/performance/common/perf_concurrent_test.py

```diff
@@ -0,0 +1,372 @@
+"""Test utilities for writing performance tests
+
+Define ConcurrentTestCase as parent class of writting Subscribe by AK,
+Subscribe by register and attach, Subscription Deletion.
+
+"""
+
+from robottelo.common import ssh
+
+import unittest
+import logging
+import csv
+
+from tests.foreman.performance.common.perf_thread \
+    import SubscribeAKPerfThread
+from tests.foreman.performance.common.perf_thread \
+    import SubscribeAttachPerfThread
+from tests.foreman.performance.common.perf_thread \
+    import DeletePerfThread
+from tests.foreman.performance.common.perf_stat \
+    import generate_stat_for_concurrent_thread
+
+
+class ConcurrentTestCase(unittest.TestCase):
+    """Parent Class of concurrent system subscription and deletion"""
+
+    @classmethod
+    def setUpClass(cls):
+        cls.logger = logging.getLogger('robottelo')
+        cls.logger.setLevel(logging.DEBUG)
+
+        """general running parameters"""
+        cls.num_threads = [1, 2, 4, 6, 8, 10]
+        cls.num_buckets = 10  # number of buckets in stat file;
+
+    def setUp(self):
+        self.logger.debug("Running test %s/%s",
+                          type(self).__name__, self._testMethodName)
+
+    @classmethod
+    def set_file_handler(cls):
```

Comment:
This is already done by the configuration. All logging will be stored on robottelo.log file on the root project dir.